### PR TITLE
Improve midFielder coach

### PR DESCRIPTION
**Pull request by RobZelluf**
Does not yet have standing free, only harassing and blocking the passline

### CODE REVIEW

File: roboteam_ai/src/analysis/DecisionMaker.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ PlayStyle DecisionMaker::getRecommendedPlayStyle(BallPossession possession) {
           {{4, 0, 0}, {3, 0, 1}, {2, 0, 2}, {2, 0, 2}, {2, 0, 2}}, // 4
           {{4, 0, 1}, {4, 0, 1}, {3, 0, 2}, {3, 0, 2}, {3, 0, 2}}, // 5
           {{5, 0, 1}, {4, 0, 2}, {4, 0, 2}, {3, 0, 3}, {3, 0, 3}}, // 6
-          {{5, 0, 2}, {5, 0, 2}, {5, 0, 2}, {4, 0, 3}, {4, 0, 3}}, // 7 // Normal
+          {{5, 1, 1}, {4, 1, 2}, {3, 2, 2}, {3, 1, 3}, {2, 2, 3}}, // 7 // Normal
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/analysis/DecisionMaker.cpp]: reeeeee from left to right midfielders: 1, 1, 2, 1, 2 is not logical

File: roboteam_ai/src/coach/midField/MidFieldCoach.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            return HARASS_OFFENSIVE;
+        } else {
+            return HARASS_DEFENSIVE;
+        }
+    }
+}
+
+MidFieldCoach::Target MidFieldCoach::getBall(RobotPtr &thisRobot, const RobotPtr& opponent) {
+    Target target;
+    target.targetRobot = opponent->id;
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/coach/midField/MidFieldCoach.cpp]: target target Tar getTar get Target target

Comment (ThijsLuttikhuis) [reply roboteam_ai/src/coach/midField/MidFieldCoach.cpp]: but it's okay

File: roboteam_ai/src/coach/midField/MidFieldCoach.cpp [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    auto ball = world::world->getBall();
+    BallPossession ballPossession;
+    auto possession = ballPossession.getPossession();
+
+    // Check if the opponent is left of us (is being offensive)
+    if (opponent->pos.x < thisRobot->pos.x) {
+
+        // Check if ball is on our left
+        if (ball->pos.x < thisRobot->pos.x) {
+            if(possession == BallPossession::OURBALL) {
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/coach/midField/MidFieldCoach.cpp]: too many if else if elses
make functions for this

File: roboteam_ai/src/coach/midField/MidFieldCoach.cpp [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    return target;
+}
+
+MidFieldCoach::RobotPtr MidFieldCoach::findRobotToHarass(const RobotPtr& thisRobot) {
+    RobotPtr closestRobot = nullptr;
+    auto shortestDistance = DBL_MAX;
+
+    // Loop over all opponents to find a opponent to harass
+    for(const auto &opponent : world::world->getThem()) {
+        // If opponent is not valid, ignore it
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/coach/midField/MidFieldCoach.cpp]: can be functions

**Review by ThijsLuttikhuis — CHANGES_REQUESTED** [review]
looks fine, but you should make smaller functions

**Review by ThijsLuttikhuis — APPROVED** [review]
needs some improvements, but as a first version its fine

File: roboteam_ai/src/skills/MidFieldHarasser.cpp [hunk 5]
```diff
@@ -28,39 +27,43 @@ Skill::Status MidFieldHarasser::onUpdate() {
...
 }
 
 
 Vector2 MidFieldHarasser::getHarassTarget() {
-    auto harassTarget =  coach::g_harassRobotCoach.getHarassPosition(robot, myIndex);
-    robotBeingHarassed = harassTarget.harassRobot;
-    return harassTarget.harassPosition;
-}
+    auto harassTarget = coach::g_midFieldCoach.getTargetPosition(robot);
+    std::cout << "Robot" << robot->id << " harassing robot " << harassTarget.targetRobot << std::endl;
```
Comment (Barisimre) [inline roboteam_ai/src/skills/MidFieldHarasser.cpp]: CRY

**Review by Barisimre — CHANGES_REQUESTED** [review]

**Pull request merged**